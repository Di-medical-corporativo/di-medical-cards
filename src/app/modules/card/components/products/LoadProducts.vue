<template>
  <div class="products__load">
    <button class="products__load__button" @click="loadMoreProducts">Cargar m√°s</button>
  </div>
</template>

<script>
import { useStore } from 'vuex'

export default {
  setup () {
    const store = useStore()

    return {
      loadMoreProducts: async () => {
        return await store.dispatch('card/getAllProducts')
      }
    }
  }
}
</script>

<style>

</style>
