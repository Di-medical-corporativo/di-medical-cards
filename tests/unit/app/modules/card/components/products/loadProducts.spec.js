import { shallowMount } from '@vue/test-utils'
import LoadProducts from '../../../../../../../src/app/modules/card/components/products/LoadProducts.vue'
import { products } from '../../../../../mock-data/testProduts'

jest.mock('vue-router', () => ({
  useRouter: jest.fn(() => ({
    push: jest.fn()
  })),
  useRoute: jest.fn(() => ({
    params: {
      sucursal: 'sur'
    }
  }))
}))

const mockStore = {
  dispatch: jest.fn().mockReturnValue(products),
  commit: jest.fn(),
  getters: ['']
}

jest.mock('vuex', () => ({
  useStore: () => mockStore
}))

describe('Load product component', () => {
  beforeEach(() => jest.clearAllMocks())

  test('should match snapshot', () => {
    const wrapper = shallowMount(LoadProducts)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should dispatch action when button is clicked', async () => {
    const wrapper = shallowMount(LoadProducts)
    await wrapper.find('.products__load__button').trigger('click')
    expect(mockStore.dispatch).toHaveBeenCalled()
  })
})
