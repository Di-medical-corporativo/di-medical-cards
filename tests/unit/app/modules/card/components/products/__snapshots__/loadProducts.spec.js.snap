// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load product component should match snapshot 1`] = `<div class="products__load"><button class="products__load__button" style="background: rgb(33, 141, 155);"> Cargar m√°s </button></div>`;
